### PR TITLE
Upgrade/kokkos

### PR DESCRIPTION
**Pull request by vincentmr**
### Before submitting

Please complete the following checklist when submitting a PR:

- [x] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [x] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [x] Ensure that the test suite passes, by running `make test`.

- [x] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [x] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**
Kokkos v4.0.00 is out. Let's try to support it.

**Description of the Change:**
Upgrade Kokkos version in workflows. Deprecate `kokkos_config_info` and replace by `print_configuration`.

**Benefits:**
Keep up with Kokkos, bug fixes and optimisations.

**Possible Drawbacks:**
Slight change in behaviour due to `kokkos_config_info` deprecation.

**Related GitHub Issues:**
None.

### CODE REVIEW

**Comment by vincentmr** [comment]
I will merge #57 first and then come back to this PR.

File: .github/workflows/wheel_linux_x86_64.yml [hunk 1]
```diff
@@ -155,7 +155,7 @@ jobs:
             source /opt/rh/devtoolset-11/enable -y
 
           CIBW_ENVIRONMENT: |
-            PATH=/opt/rh/devtoolset-11/root/usr/bin:$PATH CMAKE_ARGS="-DPYTHON_EXECUTABLE=$(which python) -DKokkos_ENABLE_${{ matrix.exec_model }}=ON"
+            PATH=/opt/rh/devtoolset-11/root/usr/bin:$PATH CMAKE_ARGS="-DPYTHON_EXECUTABLE=$(which python)"
```
Comment (mlxd) [inline .github/workflows/wheel_linux_x86_64.yml:158]: Do we no longer need to specify which backend is being built here?

**Review by mlxd — COMMENTED** [review]
Thanks @vincentmr 
Just a few quick questions

File: .github/workflows/wheel_linux_x86_64.yml (hunk 1 above)
Comment (vincentmr) [reply .github/workflows/wheel_linux_x86_64.yml:158]: It is not needed as this is specified when building the Kokkos libs, which is now done separately.

File: pennylane_lightning_kokkos/lightning_kokkos.py [hunk 2]
```diff
@@ -57,15 +54,6 @@
         LightningKokkos_C64,
         AdjointJacobianKokkos_C128,
         AdjointJacobianKokkos_C64,
```
[the remark this answers is not in the document]
Comment (vincentmr) [reply pennylane_lightning_kokkos/lightning_kokkos.py:56]: Not sure when/how/why that moved to serialize, but seems like this is what it is :D

File: .github/workflows/wheel_linux_x86_64.yml (hunk 1 above)
Comment (vincentmr) [reply .github/workflows/wheel_linux_x86_64.yml:158]: Regarding that, we are still building Kokkos on-the-fly in the test workflows. I would like to get rid of that in another PR. Just have one workflow to build and cache Kokkos-BACKEND, then reuse the libs in all workflows that need it.

Comment (mlxd) [reply .github/workflows/wheel_linux_x86_64.yml:158]: Sounds good. For the suggested change we can still build the backend of choice with setup.py though right?

Comment (vincentmr) [reply .github/workflows/wheel_linux_x86_64.yml:158]: Yes, this is done as before passing `-DKokkos_ENABLE_BACKEND=ON` in `CMAKE_ARGS`.

Comment (vincentmr) [reply .github/workflows/wheel_linux_x86_64.yml:158]: Actually, this is still done in the GPU tests, thus proving it works.

Comment (mlxd) [reply .github/workflows/wheel_linux_x86_64.yml:158]: All good so. Thanks @vincentmr 
I'll approve.

**Review by mlxd — APPROVED** [review]

**Pull request merged**